filebeat/harvester/reader/multiline.go

```diff
@@ -113,40 +115,40 @@ func (mlr *Multiline) Next() (Message, error) {
 }
 
 func (mlr *Multiline) readFirst() (Message, error) {
-	for {
-		message, err := mlr.reader.Next()
-		if err != nil {
-			// no lines buffered -> ignore timeout
-			if err == sigMultilineTimeout {
-				continue
-			}
+	//for {
+	message, err := mlr.reader.Next()
+	if err != nil {
+		// no lines buffered -> ignore timeout
+		//if err == sigMultilineTimeout {
+		//	continue
+		//}
 
-			// pass error to caller (next layer) for handling
-			return message, err
-		}
+		// pass error to caller (next layer) for handling
+		return message, err
```

Comment:
I think it is up to the caller to decide what to do with "half" messages. One thing I'm not sure about is the state of Multiline. The state of multiline should be before the error.